packages/@aws-cdk/assert/lib/assertions/have-output.ts

```diff
@@ -0,0 +1,82 @@
+import {Assertion} from "../assertion";
+import {StackInspector} from "../inspector";
+
+class HaveOutputAssertion extends Assertion<StackInspector> {
+  constructor(private readonly outputName?: string, private readonly exportName?: string, private outputValue?: any) {
+    super();
+  }
+
+  public get description(): string {
+    return `output\
+${this.outputName ? ' with name ' + this.outputName : ''}\
+${this.exportName ? ' with export name ' + this.exportName : ''}\
+${this.outputValue ? ' with value ' + JSON.stringify(this.outputValue) : ''}\
+`;
+  }
+
+  public assertUsing(inspector: StackInspector): boolean {
+    if (!this.outputName && !this.exportName) {
+      throw new Error('At least one of [outputName, exportName] should be defined');
+    }
+    if (!('Outputs' in inspector.value)) {
+      return false;
+    }
+    return (!this.outputName || this.checkOutputName(inspector)) &&
+        (!this.exportName || this.checkExportName(inspector)) &&
+        (!this.outputValue || this.checkOutputValue(inspector));
+  }
+
+  private checkOutputName(inspector: StackInspector): boolean {
+    return !!this.outputName && (this.outputName in inspector.value.Outputs);
+  }
+
+  private checkExportName(inspector: StackInspector): boolean {
+    let outputs;
+    if (this.outputName) {
+      outputs = [inspector.value.Outputs[this.outputName]];
+    } else {
+      outputs = Object.values(inspector.value.Outputs);
+    }
+    const outputWithExport = outputs.find((output: any) => {
+      return output && output.Export && (output.Export.Name === this.exportName);
+    });
+    return !!outputWithExport;
+  }
+
+  private checkOutputValue(inspector: StackInspector): boolean {
+    let output;
+    if (this.outputName) {
+      output = inspector.value.Outputs[this.outputName];
+    } else {
+      output = Object.values(inspector.value.Outputs).find((out: any) => {
+        return out && out.Export && (out.Export.Name === this.exportName);
+      });
+    }
+    return output ?
+        JSON.stringify(output.Value) === JSON.stringify(this.outputValue) :
+        false;
+
+  }
+}
+
+/**
+ * Interface for haveOutput function properties
+ * NOTE that at least one of [outputName, exportName] should be provided
+ * @property outputName   logical id of the output
+ * @property exportName   name of the resource output to be exported for a cross-stack reference
+ * @property outputValue  value of the output
```

Comment:
Those should be moved out to each property.

Reply:
You marked this (and others) resolved, but I still see those in the current state of the code. Was there somehow a reversal happening that got the code back to its initial state?

Reply:
Hi @RomainMuller , each property description has been implemented by yourself in suggestion and looks like I've just missed to remove those 3 lines above (if I understand correctly your suggestion). 
Should be fixed now.